Домашнее задание 20 спринт

darkelf007 opened the pull request.

darkelf007 committed:
15.09.2024

Commit:
16.09.2024

Commit:
16.09.2024 fix

kekulta commented on app/src/main/java/com/android/playlistmaker/di/PlayerModule.kt:
```diff
@@ -7,5 +7,5 @@ import org.koin.dsl.module
 
 val playerModule = module {
     single <MediaPlayer> { MediaPlayer() }
-    single<AudioRepository> { AudioPlayerImpl(get()) }
```
Есть новый синтаксис: https://insert-koin.io/docs/reference/koin-core/dsl-update/

Commit:
- Удален `currentPosition` из `PlayerViewState` для избежания избыточных обновлений UI.
- Добавлен вызов `seekTo(0)` после завершения трека для сброса позиции плеера.
- Оптимизированы методы `pausePlayer()` и `startPlayer()` для объединения обновлений состояния и предотвращения дублирующих уведомлений наблюдателей.

Commit:
- Рефакторинг Koin
- Исправлен интерфейс поиска по шаблону figma
- Отключен полноэкранный режим
- Добавлены параметры для стилей статус бара
